pkg/endpointmanager/manager.go

```diff
@@ -264,17 +264,19 @@ func TriggerPolicyUpdates(owner endpoint.Owner) *sync.WaitGroup {
 			ep.Mutex.Lock()
 			policyChanges, err := ep.TriggerPolicyUpdatesLocked(owner, nil)
 			regen := false
-			if err == nil && policyChanges {
-				// Regenerate only if state transition succeeds
-				regen = ep.SetStateLocked(endpoint.StateWaitingToRegenerate, "Triggering endpoint regeneration due to policy updates")
+			if err == nil {
+				if policyChanges || force {
```

Comment:
why not `if err == nil && (policyChanges || force) {`?

Reply:
Fixed.